lightning/src/onion_message/messenger.rs

```diff
@@ -583,15 +630,21 @@ where
 	let (first_node_id, blinding_point) = if let Some(first_node_id) = intermediate_nodes.first() {
 		(*first_node_id, PublicKey::from_secret_key(&secp_ctx, &blinding_secret))
 	} else {
-		match destination {
-			Destination::Node(pk) => (pk, PublicKey::from_secret_key(&secp_ctx, &blinding_secret)),
-			Destination::BlindedPath(BlindedPath { introduction_node_id, blinding_point, .. }) =>
-				(introduction_node_id, blinding_point),
+		match &destination {
+			Destination::Node(pk) => (*pk, PublicKey::from_secret_key(&secp_ctx, &blinding_secret)),
+			Destination::BlindedPath(BlindedPath { introduction_node, blinding_point, .. }) => {
+				match introduction_node {
+					IntroductionNode::NodeId(pubkey) => (*pubkey, *blinding_point),
+					IntroductionNode::DirectedShortChannelId(..) => {
+						return Err(SendError::UnresolvedIntroductionNode);
```

Comment:
> Alternatively, we might want to prefer a public function that requires a graph that wraps a `pub(crate)` method that expects it to be resolved.

Hmm... this could be a problem if the caller doesn't have a `NetworkGraph`. @Evanfeenstra Would this be an issue for your use case?

Reply:
Included in the latest push.